pkg/sql/pg_catalog.go

```diff
@@ -1811,6 +1812,19 @@ https://www.postgresql.org/docs/9.5/catalog-pg-index.html`,
 		return forEachTableDesc(ctx, p, dbContext, hideVirtual, /* virtual tables do not have indexes */
 			func(db catalog.DatabaseDescriptor, sc catalog.SchemaDescriptor, table catalog.TableDescriptor) error {
 				tableOid := tableOid(table.GetID())
+
+				// Generate a map to determine the position of a column ID for creating
+				// the indkey.
+				sortedColumns := make([]catalog.Column, len(table.PublicColumns()))
+				copy(sortedColumns, table.PublicColumns())
+				sort.SliceStable(sortedColumns, func(i, j int) bool {
+					return sortedColumns[i].GetPGAttributeNum() < sortedColumns[j].GetPGAttributeNum()
+				})
+				colToPosition := make(map[descpb.ColumnID]descpb.ColumnID)
```

Comment:
i'm not sure if mapping to positin in this slice is correct. according to PG docs, `indkey` is supposed to reference `pg_attribute.attnum`

Reply:
Ah, I missed that. @rafiss I think we need the same logic in pg_attribute, so that pg_index lines up then:

"This is an array of indnatts values that indicate which table columns this index indexes. For example, a value of 1 3 would mean that the first and the third table columns make up the index entries. Key columns come before non-key (included) columns. A zero in this array indicates that the corresponding index attribute is an expression over the table columns, rather than a simple column reference."

Does that sound good to you?

Reply:
is there a way we can make PGAttributeNum in the descriptor be what we want?

right now the comment on it is:
```
	// PGAttributeNum must be accessed through the accessor, since it is set
	// lazily, it is incorrect to access it directly.
	// PGAttributeNum represents a column's number in catalog tables.
	// This only differs from ID when the Column order is swapped or
	// the ColumnDescriptor must be remade while remaining visual ordering.
	// This does not exist in TableDescriptors pre 20.2.
```

Reply:
Yeah, let me dig a bit, since need something in PostDeserialization and the write code path to fix these up.

Reply:
@rafiss Can you take a quick peak and see if this looks saner now?